File: Leetcode/C++/1295. Find Numbers with Even Number of Digits/solution.cpp
// https://leetcode.com/problems/find-numbers-with-even-number-of-digits/description/

class Solution
{
public:
    int findNumbers(vector<int> &nums)
    {
        int count = 0;
        for (auto num : nums)
        {
            int digits = 0;

            while (num > 0)
            {
                digits += 1;
                num /= 10;
            }

            if (digits % 2 == 0)
            {
                count += 1;
            }
        }

        return count;
    }
};

File: Leetcode/Python/1295. Find Numbers with Even Number of Digits/solution.py
# https://leetcode.com/problems/find-numbers-with-even-number-of-digits/description

class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        count = 0

        for num in nums:
            if self.getNumberOfDigit(num) % 2 == 0:
                count += 1
        
        return count

    def getNumberOfDigit(self, num: int) -> int:
        digit_count = 1

        while 10 <= num:
            digit_count += 1
            num //= 10
        
        return digit_count